### PR TITLE
Editorial: dfn the Pattern data structures

### PR DESCRIPTION
**Pull request by michaelficarra**
I also updated the Notation section's ID since it was *way* under-specific.

### CODE REVIEW

File: spec.html [hunk 1]
```diff
@@ -35151,27 +35151,27 @@ <h1>Pattern Semantics</h1>
...
+            A <dfn id="pattern-state" variants="States">State</dfn> is an ordered triple (_input_, _endIndex_, _captures_) where _input_ is a List of characters representing the String being matched, _endIndex_ is an integer, and _captures_ is a List of values, one for each left-capturing parenthesis in the pattern. States are used to represent partial match states in the regular expression matching algorithms. The _endIndex_ is one plus the index of the last input character matched so far by the pattern, while _captures_ holds the results of capturing parentheses. The _n_<sup>th</sup> element of _captures_ is either a Range representing the range of characters captured by the _n_<sup>th</sup> set of capturing parentheses, or *undefined* if the _n_<sup>th</sup> set of capturing parentheses hasn't been reached yet. Due to backtracking, many States may be in use at any time during the matching process.
```
Comment (bakkot) [inline spec.html:35165]: Can we rename this to `MatchState` or something, instead of needing to emu-not-ref the uses of the word "State"?

**Comment by bakkot** [comment]
cf #2486.

File: spec.html (hunk 1 above)
Comment (michaelficarra) [reply spec.html]: @bakkot Done.

**Review by bakkot — APPROVED** [review]

**Review by syg — COMMENTED** [review]
I like the MatchState change. I have some concern over Range and Continuation being too generic to be dfn'd here. CaptureRange and MatcherContinuation maybe?

**Comment by michaelficarra** [comment]
@syg Renamed.

**Review by syg — APPROVED** [review]

**Pull request merged**